src/main/java/net/minecraftforge/common/extensions/IForgeBlock.java

```diff
@@ -951,4 +952,15 @@ default BlockState getAppearance(BlockState state, BlockAndTintGetter level, Blo
     {
         return state;
     }
+
+    /**
+     * @deprecated call the method on BlockState
+     * @param state The state of this block
+     * @return the PushReaction of this state or null if the one passed into the block properties should be used
+     */
+    @Deprecated
```

Comment:
We don't deprecate these methods. I would just leave a mention in the javadoc that it should be called from BlockState#getPushReaction.